packages/hoppscotch-cli/src/utils/getters.ts

```diff
@@ -134,3 +152,98 @@ export const roundDuration = (
   duration: number,
   precision: number = DEFAULT_DURATION_PRECISION
 ) => round(duration, precision);
+
+/**
+ * Retrieves the contents of a resource (collection or environment) from a local file (export) or a remote server (workspaces).
+ *
+ * @param {GetResourceContentsParams} params - The parameters for retrieving resource contents.
+ * @param {string} params.pathOrId - The path to the local file or the ID for remote retrieval.
+ * @param {string} [params.accessToken] - The access token for authorizing remote retrieval.
+ * @param {string} [params.serverUrl] - The SH instance server URL for remote retrieval. Defaults to the cloud instance.
+ * @param {"collection" | "environment"} params.resourceType - The type of the resource to retrieve.
+ * @returns {Promise<HoppCollection | Environment>} A promise that resolves to the contents of the resource.
+ * @throws Will throw an error if the content type of the fetched resource is not `application/json`,
+ *         if there is an issue with the access token, if the server connection is refused,
+ *         or if the server URL is invalid.
+ */
+export const getResourceContents = async (
+  params: GetResourceContentsParams
+): Promise<HoppCollection | Environment> => {
+  const { pathOrId, accessToken, serverUrl, resourceType } = params;
+
+  let contents: HoppCollection | Environment | null = null;
+  let fileExistsInPath = false;
+
+  try {
+    await fs.access(pathOrId);
+    fileExistsInPath = true;
+  } catch (e) {
+    fileExistsInPath = false;
+  }
+
+  if (accessToken && !fileExistsInPath) {
+    try {
+      const hostname = serverUrl || "https://api.hoppscotch.io";
+
+      const separator = hostname.endsWith("/") ? "" : "/";
+      const resourcePath =
+        resourceType === "collection" ? "collection" : "environment";
+
+      const url = `${hostname}${separator}v1/access-tokens/${resourcePath}/${pathOrId}`;
+
+      const { data, headers } = await axios.get(url, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      });
+
+      if (!headers["content-type"].includes("application/json")) {
+        throw new Error("INVALID_CONTENT_TYPE");
+      }
+
+      contents =
+        resourceType === "collection"
+          ? transformWorkspaceCollection(data as WorkspaceCollection)
```

Comment:
in an ideal world, we should not cast here and  run a validation, because we can't trust any backend, even if its our own.

and if the lowest layer has validation, all the above layers can use the value returned with peace of mind.

but now if you don't want to do that with our timelines, its understandable.

Reply:
Fair point. For now, any errors encountered will be caught at the `catch` block within the `getResourceContents` helper function where the above gets consumed. We could revisit this if the scope of usage goes beyond down the line.